src/msw/listctrl.cpp

```diff
@@ -3050,41 +3074,52 @@ bool HandleSubItemPrepaint(wxListCtrl* listctrl, LPNMLVCUSTOMDRAW pLVCD, HFONT h
     if ( hfont )
         selFont.Init(hdc, hfont);
 
-    // get the rectangle to paint
-    RECT rc;
-    wxGetListCtrlSubItemRect(hwndList, item, col, LVIR_BOUNDS, rc);
-    if ( !col && colCount > 1 )
-    {
-        // ListView_GetSubItemRect() returns the entire item rect for 0th
-        // subitem while we really need just the part for this column
-        RECT rc2;
-        wxGetListCtrlSubItemRect(hwndList, item, 1, LVIR_BOUNDS, rc2);
-        rc.right = rc2.left;
-    }
+    // there's some padding on both sides of the column item
+    rc.left += PADDING_LEFT_SIDE;
+    rc.right -= PADDING_RIGHT_SIDE;
+
+    const int availableWidth = wxMax(rc.right - rc.left, 0);
 
-    if ( !col && listctrl->HasCheckBoxes() )
+    if ( !col && listctrl->HasCheckBoxes() && availableWidth > 0 )
     {
         const HIMAGELIST himl = ListView_GetImageList(hwndList, LVSIL_STATE);
 
         if ( himl && ImageList_GetImageCount(himl) == 2 )
         {
-            int cbX, cbY, cbWidth, cbHeight;
+            rc.left += GAP_BEFORE_CHECKBOX;
 
+            int cbWidth, cbHeight;
             ImageList_GetIconSize(himl, &cbWidth, &cbHeight);
-            cbX = listctrl->FromDIP(GAP_BETWEEN_CHECKBOX_AND_TEXT);
-            cbY = rc.top + ((rc.bottom - rc.top) / 2 - cbHeight / 2);
+
+            // center checkbox vertically
+            int cbY = rc.top + ((rc.bottom - rc.top) / 2 - cbHeight / 2);
+
+            // prevent drawing checkbox farther than the column width
+            cbWidth = wxClip(cbWidth, 0, availableWidth);
```

Comment:
availableWidth isn't the current remaining space that's available in the column, since we've added GAP_BEFORE_CHECKBOX to it already. The checkbox is now able to overflow the next column by 4 pixels.

Reply:
Oops, thanks, I've indeed missed it. But how to fix this depends on the fix for the comment above, i.e. I would just replace `availableWidth > 0` with `availableWidth > GAP_BEFORE_CHECKBOX` and use `availableWidth - GAP_BEFORE_CHECKBOX` here, but not if we remove that check completely...

Reply:
What about the latest version? I've removed `availableWidth` to get a bit closer back to the original version, but I've also added returns to make sure we don't try drawing anything which is certain not to fit.